pkg/authorization/apis/authorization/validation/validation.go

```diff
@@ -527,3 +528,67 @@ func ValidateRoleBindingRestrictionServiceAccount(sa *authorizationapi.ServiceAc
 
 	return allErrs
 }
+
+func ValidateAccessRestriction(obj *authorizationapi.AccessRestriction) field.ErrorList {
+	allErrs := validation.ValidateObjectMeta(&obj.ObjectMeta, false, validation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateAccessRestrictionSpec(&obj.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+func ValidateAccessRestrictionSpec(spec *authorizationapi.AccessRestrictionSpec, fld *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(spec.MatchAttributes) == 0 {
+		allErrs = append(allErrs, field.Required(fld.Child("matchAttributes"), "must supply at least one policy rule"))
+	}
+
+	for i, rule := range spec.MatchAttributes {
+		allErrs = append(allErrs, rbacvalidation.ValidatePolicyRule(rule, false, fld.Child("matchAttributes").Index(i))...)
+	}
+
+	if len(spec.AllowedSubjects) == 0 && len(spec.DeniedSubjects) == 0 {
+		allErrs = append(allErrs, field.Required(fld.Child("allowedSubjects"), "either allowedSubjects or deniedSubjects must be specified"))
+	}
+
+	if len(spec.AllowedSubjects) != 0 && len(spec.DeniedSubjects) != 0 {
+		allErrs = append(allErrs, field.Invalid(fld.Child("deniedSubjects"), "<omitted>", "both allowedSubjects and deniedSubjects cannot be specified"))
+	}
+
+	for i, subjectMatcher := range spec.AllowedSubjects {
+		allErrs = append(allErrs, ValidateSubjectMatcher(subjectMatcher, fld.Child("allowedSubjects").Index(i))...)
+	}
+
+	for i, subjectMatcher := range spec.DeniedSubjects {
+		allErrs = append(allErrs, ValidateSubjectMatcher(subjectMatcher, fld.Child("deniedSubjects").Index(i))...)
+	}
+
+	return allErrs
+}
+
+func ValidateSubjectMatcher(subjectMatcher authorizationapi.SubjectMatcher, fld *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if subjectMatcher.UserRestriction == nil && subjectMatcher.GroupRestriction == nil {
+		allErrs = append(allErrs, field.Invalid(fld.Child("userRestriction"), "<omitted>", "either userRestriction or groupRestriction must be specified"))
+	}
+
+	if subjectMatcher.UserRestriction != nil && subjectMatcher.GroupRestriction != nil {
+		allErrs = append(allErrs, field.Invalid(fld.Child("groupRestriction"), "<omitted>", "both userRestriction and groupRestriction cannot be specified"))
```

Comment:
Why can't I specify both?

Reply:
Mimics RBR and you can just specify multiple `SubjectMatcher`s in the same `AccessRestriction` if you want both (in an `OR` relationship - `AND` can be achieved via multiple `AccessRestriction`s).

Reply:
> AND can be achieved via multiple AccessRestrictions).

Example?  I don't think I care about the use-case, but it seems like multiple AccessRestrictions would be an OR, not an AND